links/gql_http_link/lib/src/gql_client.dart

```diff
@@ -0,0 +1,114 @@
+import "package:collection/collection.dart";
+
+class GqlClientRequest {
+  final String method;
+  final Uri uri;
+  final Map<String, dynamic> body;
+  final Map<String, String> headers;
+  final Map<String, dynamic> files;
+  final Map<String, List<String>> fileMapping;
+
+  GqlClientRequest({
```

Comment:
Mart the required fields as `@required` or add default values
method, uri, body, headers as required  (not sure about file and fileMapping)